server/functional-test/src/test/java/org/zanata/feature/editor/TranslateTextTest.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2014, Red Hat, Inc. and individual contributors as indicated by the
+ * @author tags. See the copyright.txt file in the distribution for a full
+ * listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This software is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
+ * site: http://www.fsf.org.
+ */
+package org.zanata.feature.editor
+
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.experimental.categories.Category
+import org.zanata.feature.Trace
+import org.zanata.feature.testharness.TestPlan.DetailedTest
+import org.zanata.feature.testharness.ZanataTestCase
+import org.zanata.page.webtrans.EditorPage
+import org.zanata.util.CleanDocumentStorageRule
+import org.zanata.util.TestFileGenerator
+import org.zanata.util.ZanataRestCaller
+import org.zanata.workflow.BasicWorkFlow
+import org.zanata.workflow.LoginWorkFlow
+import org.zanata.workflow.ProjectWorkFlow
+
+import java.io.File
+
+import org.assertj.core.api.Assertions.assertThat
+import org.zanata.util.FunctionalTestHelper.assumeFalse
+
+/**
+ * See the unit tests in package org.zanata.adapter for other file types.
+ * @author Damian Jansen [djansen@redhat.com](mailto:djansen@redhat.com)
+ */
+@Category(DetailedTest::class)
+class TranslateTextTest : ZanataTestCase() {
+
+    @get:Rule
+    var documentStorageRule = CleanDocumentStorageRule()
+
+    private lateinit var zanataRestCaller: ZanataRestCaller
+
+    private val testFileGenerator = TestFileGenerator()
+
+    @Before
+    fun before() {
+        zanataRestCaller = ZanataRestCaller()
+        BasicWorkFlow().goToHome().deleteCookiesAndRefresh()
+        assumeFalse(
+                "",
+                File(CleanDocumentStorageRule.getDocumentStoragePath() +
+                        "${File.separator}documents${File.separator}").exists())
```

Comment:
I know this is probably pre-existing logic, but wouldn't it be better just to delete the the directory, or simply fail the test? 

If this condition fails, I think the test will be skipped with little or no warning. That doesn't seem right.

Reply:
Indeed, it's unnecessary - the rule throws if a delete fails.